compiler/riscv/codegen/RVInstruction.cpp

```diff
@@ -495,15 +495,15 @@ uint8_t *TR::JtypeInstruction::generateBinaryEncoding() {
          }
       else
          {
-         TR_ASSERT(0, "Non-recursive calls not (yet) supported");
+         offset = (uint8_t*)getSymbolReference()->getMethodAddress() - cursor;
          }
       }
    else
       {
-      uintptr_t destination = (uintptr_t)(getLabelSymbol()->getCodeLocation());
+      uint8_t* destination = (uint8_t*)(getLabelSymbol()->getCodeLocation());
```

Comment:
Please make this a `reinterpret_cast` as the compiler is starting to adopt the C++ casts exclusively.